plugins/user-authenticators/saml2/src/org/apache/cloudstack/api/command/SAML2LoginAPIAuthenticatorCmd.java

```diff
@@ -303,9 +308,13 @@ public String authenticate(final String command, final Map<String, Object[]> par
                 }
 
                 if (userAccount == null || userAccount.getExternalEntity() == null || !_samlAuthManager.isUserAuthorized(userAccount.getId(), issuer.getValue())) {
```

Comment:
This method is already quite big. What about extracting what you added lines 310-318 to a method?
This enables, docs, test case and a better and clear code to read (in my opinion).